Sources/Smithy/Logging/LogAgent.swift

```diff
@@ -5,7 +5,7 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-public protocol LogAgent {
+public protocol LogAgent: Sendable {
```

Comment:
See new warnings related to TestLogger.  They should be easily fixed.

Reply:
Good catch, I think there's an instance in aws-sdk-swift unit tests also where this needs to be addressed.